### ISSUE TITLE
UserID

### ISSUE THREAD

**Issue by daremeapp** [dshukertjr/spot#49]
I'm adding some custom features, but are most used to firebase auth. Can you help me with the how to identify the user in a new page? Ps. Love your work!

**Comment by dshukertjr** [comment]
Hi @daremeapp!

Thank you for the kind words! It really means a lot to me!

Sure, I can try to help you out! What exactly do you mean by "identify the user in a new page"? Do you mean to get their user_id?

**Comment by daremeapp** [comment]
Yes, correct. I'm use to work with firebase, so going out to supabase is a bit confusing for me. But I said before, I just love what your have done her.

**Comment by daremeapp** [comment]
To try to explain easier, this is the model I'm used to with firebare to identify the user: 


class _State extends State<> {
  // var _isLoading = false;
  final _auth = FirebaseAuth.instance;

  void _x(
    String 
    String
    String 
    File 
  ) async {
    try {

      final ref = FirebaseStorage.instance
          .ref()
          .child('')
          .child(‘’);

      await ref.putFile(thumbnail).whenComplete(() => null);
      final url = await ref.getDownloadURL();
      await FirebaseFirestore.instance
          .collection('')
          .doc(_auth.currentUser.uid)
          .set({
 ,
      });
    } catch (err) {
      print(err);

 
    }
  }

**Comment by dshukertjr** [comment]
Sorry for the late reply. I am moving to another location, and was busy with that. 

In Supabase, you can get the `user` and `user_id` like this:

```dart
final user = supabase.auth.user();
final userId = user.id;
```

Is this what you were looking for?

Also, if you go to [Supabase discussion page](https://github.com/supabase/supabase/discussions), you can ask any Supabase related questions and someone should be able to answer them in no time!
